### PR TITLE
Add template for creating a private endpoint with a search service

### PR DESCRIPTION
**Pull request by mattmsft**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by mattmsft** [comment]
@HeidiSteen

**Comment by azure-quickstarts** [comment]
@mattmsft - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.search/azure-search-create-private-endpoint/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
...
+[![Deploy To Azure US Gov](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.svg?sanitize=true)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.search%2Fazure-search-create-private-endpoint%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.search%2Fazure-search-create-private-endpoint%2Fazuredeploy.json)
+
+This template creates a new Azure Cognitive Search Service with a private endpoint. The [quickstart](https://docs.microsoft.com/azure/search/search-get-started-arm) article describes how to deploy the template.
```
Comment (HeidiSteen) [inline quickstarts/microsoft.search/azure-search-create-private-endpoint/README.md:7]: service is lower case.   regarding phrasing, "agency" phrasing is a little odd.  The template doesn't create anything, and the customer isn't deploying the template.  The link should be relative, not absolute.

I would say:  Use this template to create an Azure Cognitive Search service that's configured for connections over a private endpoint. See [Quickstart: Deploy Cognitive Search using an Azure Resource Manager template](search-get-started-arm.md) for help on template deployment.

File: quickstarts/microsoft.search/azure-search-create-private-endpoint/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+If you are unclear as to what parameters are allowed you can check the [Microsoft.Search searchServices template reference](https://docs.microsoft.com/azure/templates/Microsoft.Search/searchServices).
```
Comment (HeidiSteen) [inline quickstarts/microsoft.search/azure-search-create-private-endpoint/README.md:9]: For more information about parameters, see [Microsoft.Search searchServices template reference](/azure/templates/Microsoft.Search/searchServices).

File: quickstarts/microsoft.search/azure-search-create-private-endpoint/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                1,
+                2,
+                3,
+                4,
+                6,
+                12
+            ],
+            "type": "int",
+            "metadata": {
+                "description": "Partitions allow for scaling of document count as well as faster indexing by sharding your index over multiple search units."
```
Comment (HeidiSteen) [inline quickstarts/microsoft.search/azure-search-create-private-endpoint/azuredeploy.json:49]: A search unit is defined as replica * partition so this phrasing is potentially confusing.  What do you think about: "Partitions are units of storage. Increasing partitions adds capacity and speed by spreading an index over multiple shards."

File: quickstarts/microsoft.search/azure-search-create-private-endpoint/metadata.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "type": "QuickStart",
+    "itemDisplayName": "Azure Cognitive Search service with private endpoint",
+    "description": "This template creates an Azure Cognitive Search service with a private endpoint",
```
Comment (HeidiSteen) [inline quickstarts/microsoft.search/azure-search-create-private-endpoint/metadata.json:5]: does the description need a period?

File: quickstarts/microsoft.search/azure-search-create-private-endpoint/metadata.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    "summary": "Provision an Azure Cognitive Search service with a private endpoint",
```
Comment (HeidiSteen) [inline quickstarts/microsoft.search/azure-search-create-private-endpoint/metadata.json:6]: ditto -- add a period??

**Review by HeidiSteen — APPROVED** [review]
Left some comments

**Comment by azure-quickstarts** [comment]
@mattmsft - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by mattmsft** [comment]
#sign-off

**Comment by bmoore-msft** [comment]
@mattmsft - check the pipeline results for changes that need to be made...

**Comment by HeidiSteen** [comment]
@bmoore-msft  Do you have more info as to what needs to be addressed?  These are all warnings, and it's not clear what fixes are required.

**Comment by bmoore-msft** [comment]
Take a look at the logs from the pipeline for each task that failed, details will be in those logs...

**Comment by HeidiSteen** [comment]
I see it now -- details are under the Checks tab --> https://github.com/Azure/azure-quickstart-templates/pull/12574/checks

**Comment by azure-quickstarts** [comment]
@mattmsft - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request merged**